### PR TITLE
[SPEAK] and [EDITOR] prompt idea

### PR DESCRIPTION
**Pull request by shiffman**
#28

### CODE REVIEW

**Comment by JuggernautJf** [comment]
umm i have ollama installed...
but how do you run it in the extension?

**Comment by JuggernautJf** [comment]
before
#### Imports are getting cryptic, so i suggest to add to `jsconfig.json`:
```json
{
  "compilerOptions": {
    "module": "Node16",
    "target": "ES2022",
    "checkJs": false /* Typecheck .js files. */,
    "lib": ["ES2022"],
    "baseUrl": "src",
    "paths": { 
      "@lib/*": ["src/lib/*"],
      "@util/*": ["src/util/*"]
    }
  },
  "exclude": ["node_modules"]
}
```
### then, the imports will look like:
after

File: src/lib/agent/providers/replicateProvider.js [hunk 1]
```diff
@@ -12,7 +13,7 @@ class ReplicateProvider extends ModelProvider {
 
   async query(prompt) {
     this.messageHistory.push({ role: 'user', content: prompt });
-    const output = await this.replicate.run('meta/llama-2-7b-chat', {
+    const output = await this.replicate.run('meta/llama-2-70b-chat', {
```
Comment (JuggernautJf) [inline src/lib/agent/providers/replicateProvider.js:16]: `output` is an object not a list...

**Comment by JuggernautJf** [comment]
## 🥺 How do i get it to run? (I can't figure it out...)

**Comment by dipamsen** [comment]
@JuggernautJf we have finally settled on testing with replicate for now, to run it create a config.js file and add your replicate api key in it

File: src/lib/agent/providers/replicateProvider.js (hunk 1 above)
Comment (dipamsen) [reply src/lib/agent/providers/replicateProvider.js:16]: it is infact an array:

image

image

**Comment by JuggernautJf** [comment]
> @JuggernautJf we have settled on testing with replicate for now, to run it create a config.js file and add your replicate api key in it

thing is I don't know how to get a key without being a whole company.....
is there a link which can take me in the right direction?

**Comment by supercrafter100** [comment]
You can create an account for free by just signing in using GitHub. Then generate an API key and use that. You *should* be able to generate quite some time for free.

**Pull request merged**